### PR TITLE
Add Changes and Releases section

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/3870

This adds all the logic needed to aggregate changelog entries from external sources and present them in a new top level section

### Tasks

- [x] Change terminology to use term "tenant cluster release" (https://github.com/giantswarm/giantswarm/issues/14339)

### Preview

<details>
<summary>List page top</summary>

image

</details>

<details>
<summary>List page bottom</summary>

image

</details>

<details>
<summary>List page, filtered to one category</summary>

image

</details>

<details>
<summary>Details page (Permalink)</summary>

image

</details>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,8 +12,20 @@ vendor:
...
+	docker run --rm \
+	  --volume=${PWD}/scripts/aggregate-changelogs:/workdir:ro \
+	  --volume=${PWD}/src/content/changes:/output:rw \
+	  -w /workdir \
+	  --env GITHUB_TOKEN \
+	  aggregate-changelogs \
+	  /workdir/script.py /workdir/config.yaml /output
 
-build: vendor
+build: vendor changes
```
Comment (marians) [inline Makefile:28]: This ensures that changes are pulled on every `make` execution.

File: scripts/aggregate-changelogs/config.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  giantswarm/aqua-app:
+    category: Managed Apps
+  giantswarm/efk-stack-app:
+    category: Managed Apps
+  giantswarm/kong-app:
+    category: Managed Apps
+  giantswarm/nginx-ingress-controller-app:
+    category: Managed Apps
+  giantswarm/prometheus-operator-app:
+    category: Managed Apps
```
Comment (marians) [inline scripts/aggregate-changelogs/config.yaml:54]: @cokiengchiara @oponder The goal here is to have all "official" apps included. Can you check whether this is complete? I added those that are pre-installed on Azure, AWS or KVM and those that are in the Giant Swarm Catalog.

File: src/config.toml [hunk 3]
```diff
@@ -2,6 +2,7 @@ baseurl = "https://docs.giantswarm.io"
 languageCode = "en"
 title = "Giant Swarm Documentation"
 pluralizeListTitles = false
+rssLimit = 50
```
Comment (marians) [inline src/config.toml:5]: Limits the RSS feeds to max 50 items. Default would be unlimited.

File: src/layouts/partials/feedback_reference.html [hunk 4]
```diff
@@ -1,4 +1,6 @@
+{{ with .File }}
 <div class="feedback well">
   <h5><i class="fa fa-help-outline"></i> Anything missing here?</h5>
-  <p>We welcome your <a title="open this page in GitHub" href="https://github.com/giantswarm/docs/blob/master/src/content/{{ .File.Path }}" target="_blank" rel="noopener noreferrer">pull requests</a>, feedback and all sorts of questions. You can also send us an email to <a href="mailto:support@giantswarm.io">support@giantswarm.io</a>.</p>
+  <p>We welcome your <a title="open this page in GitHub" href="https://github.com/giantswarm/docs/blob/master/src/content/{{ .Path }}" target="_blank" rel="noopener noreferrer">pull requests</a>, feedback and all sorts of questions. You can also send us an email to <a href="mailto:support@giantswarm.io">support@giantswarm.io</a>.</p>
 </div>
+{{ end }}
```
Comment (marians) [inline src/layouts/partials/feedback_reference.html:6]: Unrelated boyscouting, fixing a warning

File: src/layouts/partials/help_feedback.html [hunk 5]
```diff
@@ -1,4 +1,6 @@
+{{ with .File }}
 <div class="feedback well">
   <h5><i class="fa fa-help-outline"></i> Need help, got feedback?</h5>
-  <p>We listen in your Slack support channel. You can also reach us at <a href="mailto:support@giantswarm.io">support@giantswarm.io</a>. And of course, we welcome your <a title="open this page in GitHub" href="https://github.com/giantswarm/docs/blob/master/src/content/{{ .File.Path }}" target="_blank" rel="noopener noreferrer">pull requests</a>!</p>
+  <p>We listen in your Slack support channel. You can also reach us at <a href="mailto:support@giantswarm.io">support@giantswarm.io</a>. And of course, we welcome your <a title="open this page in GitHub" href="https://github.com/giantswarm/docs/blob/master/src/content/{{ .Path }}" target="_blank" rel="noopener noreferrer">pull requests</a>!</p>
 </div>
+{{ end }}
```
Comment (marians) [inline src/layouts/partials/help_feedback.html:6]: Unrelated boyscouting, fixing a warning

**Comment by marians** [comment]
Dear @giantswarm/sig-ux @giantswarm/sig-docs, this is a larger PR. Sorry and thanks for reviewing!

File: scripts/aggregate-changelogs/config.yaml (hunk 2 above)
Comment (oponder) [reply scripts/aggregate-changelogs/config.yaml:54]: Looks ok to me for now! Is there any rhyme or reason to the order? Can it be alphabetised for easier spotting of missing ones and changes in the future?

File: src/assets/styles/changes.sass [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+        // &:before
+        //     content: ""
+        //     width: 100%
+        //     height: 100%
+        //     position: absolute
+        //     left: 0
+        //     top: 0
+        //     background: linear-gradient(transparent 150px, white)
+        //     pointer-events: none
```
Comment (oponder) [inline src/assets/styles/changes.sass:95]: Commented out code can be removed unless you intend to add it back before merging.

**Review by oponder — APPROVED** [review]
LGTM and saw your workflow notes in the parent issue 👍  great step forward!

File: scripts/aggregate-changelogs/config.yaml (hunk 2 above)
Comment (marians) [reply scripts/aggregate-changelogs/config.yaml:54]: You're right, will order by key.

File: src/assets/styles/changes.sass (hunk 6 above)
Comment (marians) [reply src/assets/styles/changes.sass:95]: Right. I will move it to an issue for a future front end engineer :)

**Comment by marians** [comment]
With the last commit I added azure-admission-controller and aws-admission-controller so that their changes will appear in the `Control Plane Kubernetes API`.

Preview:

image

There are releases without info in azure-admission-controller, hence the empty block there.

**Comment by marians** [comment]
@puja108 Do you think it's fine to use "tenant cluster release" here as described above, even before we have https://github.com/giantswarm/giantswarm/issues/14339 done? The worst that can happen is that we have to change it and confuse a few people.

**Comment by puja108** [comment]
@marians I think it's fine. I was just thinking about the order:
1. TCR for AWS
2. AWS TCR
both make sense, but I was asking myself if when you are scanning for quick filtering, your intention as a user is to limit to your provider. Then having the provider at the start of the phrase would make your visual scan faster, WDYT?

**Comment by marians** [comment]
> @marians I think it's fine. I was just thinking about the order:
> 
> 1. TCR for AWS
> 2. AWS TCR
>    both make sense, but I was asking myself if when you are scanning for quick filtering, your intention as a user is to limit to your provider. Then having the provider at the start of the phrase would make your visual scan faster, WDYT?

Here, the form `<Provider> Tenant Cluster Releases` would result in all three provider categories not appearing together, but have other categories in-between. I know, that's a technical reason, but currently I don't know how to overcome this.

However, in terms of language I also think `Tenant Cluster Releases for AWS` reads more natural.

File: src/assets/styles/changes.sass (hunk 6 above)
Comment (marians) [reply src/assets/styles/changes.sass]: Comments are removed

**Comment by marians** [comment]
As per SIG UX feedback, the dynamic category colors weren't working well. With https://github.com/giantswarm/docs/pull/624/commits/e85fb9230eacb8acb8207a74d296bc59cad7bfea this is changed to use static colors.

<details>
<summary>Preview</summary>

image

</details>

**Comment by marians** [comment]
Permalink URLs are now adapted to contain the category name.

- `/changes/control-plane-kubernetes-api/apiextensions/v3.8.0/`
- `/changes/web-ui/happa/v1.0.3/`
- `/changes/gsctl/gsctl/0.25.0/` 🤷

**Pull request merged**